### PR TITLE
Update dependencies to resolve vulnerabilities from transitive dependencies

### PR DESCRIPTION
**Pull request by YangSan0622**
Motivation: Update dependencies for security purpose

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: serializer-deserializer/pom.xml [hunk 1]
```diff
@@ -221,6 +232,12 @@
             <groupId>com.google.protobuf</groupId>
             <artifactId>protobuf-java</artifactId>
         </dependency>
+        <!-- Temporarily adding dependency on a transitive dependency to fix security bug in underlying library. Remove when dependency is upgraded. -->
+        <dependency>
+            <groupId>org.apache.commons</groupId>
+            <artifactId>commons-collections4</artifactId>
```
Comment (blacktooth) [inline serializer-deserializer/pom.xml:238]: Do we need to remove the old collections dependency? Or is it a transitive dependency?

**Comment by blacktooth** [comment]
Does the dependency tree show that these vulnerable dependencies are removed?

**Comment by YangSan0622** [comment]
Intellij has a dependency checker(powered by Chekmarx), I confirmed the these vulnerabilities disappeared there

**Review by blacktooth — APPROVED** [review]

File: serializer-deserializer/pom.xml (hunk 1 above)
Comment (YangSan0622) [reply serializer-deserializer/pom.xml:238]: it is a transitive dependency, I have excluded it

**Pull request merged**